votor/src/certificate_pool_service.rs

```diff
@@ -275,7 +275,10 @@ impl CertificatePoolService {
                     return Ok(());
                 }
             }
-            stats.maybe_report();
+            if stats.maybe_report() {
+                // Stats were reported, report the certificate pool stats
+                cert_pool.report_stats();
```

Comment:
if we're going to decouple the service and pool stats, might as well fully decouple them and just call something like this in succession:
```
stats.maybe_report();
cert_pool.maybe_report();
```

Reply:
Done.